### PR TITLE
feat(vercel): Add support for image optimization API

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

This adds support for the Vercel Image Optimization API:tm: to the Vercel adapter. An image service for `astro:assets` using the API is additionally provided behind a config flag `imageService`. In dev, a slightly modified base Squoosh service is used instead.

Using the image service is optional, you can just provide [an image config](https://vercel.com/docs/build-output-api/v3/configuration#images), like such :  `adapter: vercel({imageConfig: {sizes: [...]}})` and use the API directly. The image service handles it all for you though, including an hopefully helpful default configuration

## Testing

Added tests. We don't fully E2E test it, but we make sure to generate the proper URLs at least

## Docs

Added docs

### CODE REVIEW

**Comment by Princesseuh** [comment]
!preview vercel-image

**Comment by Princesseuh** [comment]
!preview vercel-image

**Comment by Princesseuh** [comment]
!preview vercel-image

**Comment by Princesseuh** [comment]
!preview vercel-image

**Comment by Princesseuh** [comment]
!preview vercel-image

**Comment by Princesseuh** [comment]
!preview vercel-image

**Review by matthewp — APPROVED** [review]

File: packages/integrations/vercel/README.md [hunk 1]
```diff
@@ -108,6 +108,61 @@ export default defineConfig({
...
+  })
+});
+```
+
+### imageService
+
+**Type:** `boolean`<br>
+**Available for:** Edge, Serverless, Static
+
+When enabled, a [Image Service](https://docs.astro.build/en/reference/image-service-reference/) powered by the Vercel Image Optimization API will be automatically configured and used in production. In development, a built-in Squoosh-based service will be used instead.
```
Comment (sarah11918) [inline packages/integrations/vercel/README.md:138]:
```suggestion
When enabled, an [Image Service](https://docs.astro.build/en/reference/image-service-reference/) powered by the Vercel Image Optimization API will be automatically configured and used in production. In development, a built-in Squoosh-based service will be used instead.
```

**Review by sarah11918 — COMMENTED** [review]
Just a couple of minor nits from me, @Princesseuh , and just checking on the "Added in" field since it exists!

Note: I checked on Vercel's page, and it looks like the name of the API is "Image Optimization" which means you have the choice of saying "Vercel's Image Optimization API" or "the Vercel Image Optimization API" whenever it comes up. But it should be one or the other: possessive, or with a "the" in front of it. I made a suggestion below for the one you didn't use later (for sentence variety!) but it's totally your call.

File: packages/integrations/vercel/README.md [hunk 2]
```diff
@@ -108,6 +108,61 @@ export default defineConfig({
 });
 ```
 
+### imageConfig
+
+**Type:** `VercelImageConfig`<br>
+**Available for:** Edge, Serverless, Static
+
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply packages/integrations/vercel/README.md:115]: They should, good catch!

File: packages/integrations/vercel/README.md [hunk 3]
```diff
@@ -108,6 +108,61 @@ export default defineConfig({
...
+Configuration to use for [Vercel Image Optimization API](https://vercel.com/docs/concepts/image-optimization). See [this page](https://vercel.com/docs/build-output-api/v3/configuration#images) for a complete list of supported parameters.
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply packages/integrations/vercel/README.md:116]: I was desperately trying to avoid the "this page" pattern and couldn't think of anything, thank you!

**Review by natemoo-re — APPROVED** [review]
Code looks good!

**Review by Yan-Thomas — APPROVED** [review]
Docs LGTM!

**Pull request merged**